### PR TITLE
Add a context with deadline for 60 seconds

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: canary.go [hunk 1]
```diff
@@ -20,6 +21,9 @@ type TunnelTest struct {
 }
 
 func main() {
+	_, cancel := context.WithDeadline(context.Background(), time.Now().Add(60*time.Second))
```
Comment (nicpottier) [inline canary.go]: This doesn't do anything alone.. we need to pass the context created to the various functions. (Dial, Command etc..):
  https://golang.org/pkg/net/#Dialer.DialContext
  https://golang.org/pkg/os/exec/#CommandContext

**Review by nicpottier — CHANGES_REQUESTED** [review]

**Comment by nicpottier** [comment]
Also probably makes sense for those contexts to be per connection.. IE, 60 seconds each.

File: canary.go (hunk 1 above)
Comment (norkans7) [reply canary.go]: Thanks that makes more sense now

File: canary.go [hunk 2]
```diff
@@ -38,17 +40,20 @@ func main() {
 	}
 
 	for _, tunnel := range tunnels {
-		conn, err := net.DialTimeout("tcp", tunnel.Host, time.Second*10)
+		var d net.Dialer
+		ctx, _ := context.WithDeadline(context.Background(), time.Now().Add(1*time.Second))
```
Comment (nicpottier) [inline canary.go]: `WithTimeout` is the more natural way of doing this. You still want the defer cancel as you had before too. I'd say setting this to 1 minute per is fine, we just want to make sure it doesn't hang forever.

**Review by nicpottier — CHANGES_REQUESTED** [review]

File: canary.go (hunk 2 above)
Comment (norkans7) [reply canary.go]: ohhh the 1 second was trying something and forgot to revert that

**Review by nicpottier — APPROVED** [review]

**Pull request merged**